NYPL/dgx-header-component
# Log out with loading iframe

This PR adds the function to completely log out the users from Encore. For doing so, we need to hit the endpoint of log out button. And this so far is the only way we can do and know to completely log out the users.

The best way to achieve this on client is to do it with AJAX, however, III doesn't allow us to hit it with different domains (CORS policy), therefore, to bypass it, the new function load an iframe with its `src` attribute as the log out endpoint. Once the iframe is loaded, the endpoint will be hit to log out the user, and then, we delete the iframe.

More detailed discussions and experiments can be reviewed in this ticket,
https://jira.nypl.org/browse/SCC-1314

The way to test it, 

1) Visit browse.nypl.org and log in.
2) Once log in, manually set the cookie for the last time logging in Encore. Type this command in your browser's console. `document.cookie="ENCORE_LAST_VISITED=" + Date.now() + "; path=/; domain=.nypl.org";`
3) Open another window and run the PR's branch locally. Once it is loaded, it will set the timer to count down the time before log out.
4) Do not visit any pages under browse.nypl.org 's domain. After 30 mins, the iframe will be loaded on the second window, and it should log out the user from the first window.
5) Click `My Account` link in the first window, it should ask you to log in again.
6) By managing the cookie's value, we can make up the passing time to reduce the waiting time before the iframe is going to be loaded. For example, using `document.cookie="ENCORE_LAST_VISITED=" + (Date.now() - 1790000) + "; path=/; domain=.nypl.org";`  should make you wait for only 10 seconds instead of 30 mins.
